x-pack/plugins/endpoint/server/routes/endpoints.test.ts

```diff
@@ -120,4 +120,82 @@ describe('test endpoint route', () => {
     expect(endpointResultList.request_page_index).toEqual(10);
     expect(endpointResultList.request_page_size).toEqual(10);
   });
+
+  describe('Endpoint Details route', () => {
+    it('should return 404 on no results', async () => {
```

Comment:
@charlie-pichette I believe this tests the case where the endpoint doesn't exist

Reply:
Do we all agree it should be 404, I think so going by what I have seen about REST the appropriate response is missing data should be 404.

Reply:
@pzl what is 'BADID'?  That seems to me to be a malformed id instead of an id that doesn't exist.  If we are not doing any validation on the id, such as if it is supposed to be a UUID that is a valid UUID format, then this seems ok.

Reply:
We are not doing any validation on the endpoint id being submitted.  I'm fine with this test.